client/src/pages/FilmDetails.tsx

```diff
@@ -34,6 +63,36 @@ export function FilmDetailPage() {
     return <h3 style={{ color: 'white' }}>Film not found!</h3>;
   }
 
+  async function handleAddToWishlist() {
+    if (!user) {
+      alert('Please sign up or log in to save!');
+    } else {
```

Comment:
To make this cleaner, use "fail fast" and return inside the `if (!user)` block. Then you can remove a level of indenting because you'll no longer need the `else`.